src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -31,14 +37,31 @@ object Modifier {
   */
 trait Period extends TimeExpression with TemporalAmount
 
-case class SimplePeriod(unit: TemporalUnit, n: Number, modifier: Modifier) extends Period {
-  override def addTo(temporal: Temporal): Temporal = ???
+case class SimplePeriod( unit: TemporalUnit, n: Number, modifier: Modifier) extends Period {
 
-  override def get(unit: TemporalUnit): Long = ???
+  val number = n match {
+    case n:IntNumber => n.getValue()
+    case n:Number => ???
+  }
 
-  override def subtractFrom(temporal: Temporal): Temporal = ???
+  override def addTo(temporal: Temporal): Temporal = {
+    return temporal.plus( number , unit )
+  }
 
-  override def getUnits: util.List[TemporalUnit] = ???
+  override def get(unit: TemporalUnit): Long = {
+    if ( unit == this.unit )
+      return number
+    else
+      throw new UnsupportedTemporalTypeException("")
+  }
+
+  override def subtractFrom(temporal: Temporal): Temporal = {
+    return temporal.minus( number, unit )
+  }
+
+  override def getUnits: java.util.List[TemporalUnit] = {
+    return ( List( unit ) ).asJava
```

Comment:
The outer parentheses are unnecessary.
Also, rather than using JavaConverters's .asJava, use java.util.Collections.singletonList(). There's some overhead to JavaConverters, so if there's a single methods in standard Java that does the same thing, better to use that.